src/Application/Borrows/Commands/BorrowDocument.cs

```diff
@@ -0,0 +1,138 @@
+using Application.Common.Exceptions;
+using Application.Common.Interfaces;
+using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using Domain.Entities.Physical;
+using Domain.Statuses;
+using FluentValidation;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+namespace Application.Borrows.Commands;
+
+public class BorrowDocument
+{
+    public class Validator : AbstractValidator<Command>
+    {
+        public Validator()
+        {
+            RuleLevelCascadeMode = CascadeMode.Stop;
+
+            RuleFor(x => x.Reason)
+                .MaximumLength(512).WithMessage("Reason cannot exceed 512 characters.");
+
+            RuleFor(x => x.BorrowFrom)
+                .GreaterThan(DateTime.Now).WithMessage("Borrow date cannot be in the past.")
+                .Must((command, borrowTime) => borrowTime < command.BorrowTo).WithMessage("Due date cannot be before borrow date.");
+
+            RuleFor(x => x.BorrowTo)
+                .GreaterThan(DateTime.Now).WithMessage("Due date cannot be in the past.");
```

Comment:
Is there a way to check BorrowFrom < BorrowTo?

Reply:
2 lines above this line